LLMonFHIR/FHIRInterpretation/UserStudy/UserStudyViewModel+ProcessingState.swift

```diff
@@ -0,0 +1,45 @@
+//
+// This source file is part of the Stanford Spezi project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+
+
+extension UserStudyChatViewModel {
+    enum ProcessingState: Equatable {
+        case processingSystemPrompts
+        case processingFunctionCalls(progress: Double, currentCall: Int, totalCalls: Int)
```

Comment:
Couldn't we compute the progress from the other values? Then we don't need to set it every time and we could go ahead and calculate it in the progress computed properly below.